common.py

```diff
@@ -662,6 +662,6 @@ def get_future_state(future: Future[bool]) -> str:
 
         time.sleep(30)
 
-    failed = next((name for name, result in state.items() if not result), None)
+    failed = next((name for name, future in futures if not future.result()), None)
```

Comment:
I find this hard to read. Maybe that's why a bug slipped in here too. Can we write it as?

```
failed_states = {k: v for (k, v) in state.items() if v == "failed"]
if failed_states:
     # log .. 
```